probe-rs/src/probe/cmsisdap/mod.rs

```diff
@@ -1218,6 +1267,51 @@ impl SwoAccess for CmsisDap {
     }
 }
 
+impl RawJtagIo for CmsisDap {
+    fn state(&self) -> &JtagDriverState {
+        &self.jtag_driver_state
+    }
+    fn state_mut(&mut self) -> &mut JtagDriverState {
+        &mut self.jtag_driver_state
+    }
+    fn shift_bit(&mut self, tms: bool, tdi: bool, capture: bool) -> Result<(), DebugProbeError> {
+        self.jtag_driver_state.state.update(tms);
+        if let Some(sequence) = self.jtag_sequences.last_mut() {
+            if sequence.append(tms, tdi, capture).is_ok() {
+                return Ok(());
+            }
+        }
+        self.jtag_sequences.push(JtagSequence::new(
+            1,
+            capture,
+            tms,
+            [u8::from(tdi), 0, 0, 0, 0, 0, 0, 0],
+        )?);
+        Ok(())
+    }
+    fn read_captured_bits(&mut self) -> Result<BitVec<u8, Lsb0>, DebugProbeError> {
+        let mut capture = BitVec::<u8, Lsb0>::new();
+        // We can transfer up to 255 sequences in one dap jtag-sequence command. But in some riscv chip there is some bug, thus I set up to 3 sequence in once jtag-sequence command transfer.
```

Comment:
What do you mean there is a bug in some RISC-V chips? How is chunking a probe operation into groups of 3 affect what happens on the target?

Reply:
In theory, hs daplink can have a maximum of 255 sequences at a time, and a maximum of 512 bytes plus a one-byte command and a one-byte sequence count. However, in the hpm5301 test, more than 10 sequences in a command are unstable. From the log, the probe read dm register and then detach. 
![17b25abedd8219d5e38eb2e73f7ba4d5](https://github.com/probe-rs/probe-rs/assets/172099550/262f430c-660e-40aa-ba68-bb0c80af8aee)

Reply:
Is this a bug in our risc-v implementation?

Reply:
I'm not sure.

Reply:
I analyzed the log yesterday, and the error occurred when probe read back the download algorithm.
probe-rs\src\architecture\riscv\dtm\jtag_dtm.rs: 80

Reply:
> I analyzed the log yesterday, and the error occurred when probe read back the download algorithm. probe-rs\src\architecture\riscv\dtm\jtag_dtm.rs: 80

I tried both Jlink and CMSIS-DAP probe to connect to a HPM5360 RISC-V chip, Jlink works fine, CMSIS-DAP brokes down. I think the problem does not lie in architecture layer, probably probe support layer.
I'm trying to analyze the CMSIS-DAP related issues.

Reply:
With Jlink the JTAG sequences are okay, but with a CMSIS-DAP, there's no waveform.